nixos/modules/system/boot/kernel.nix

```diff
@@ -54,9 +54,76 @@ in
     };
 
     boot.kernelParams = mkOption {
+      type = let
+        coerce = list: let
+          mkItem = item: let
+            splitted = builtins.match "([^=]*)=(.*)" item;
+          in if splitted == null then nameValuePair item true
+             else nameValuePair (head splitted) (last splitted);
+        in listToAttrs (map mkItem list);
+        targetType = with types; attrsOf (either (either bool int) str);
+      in with types; coercedTo (listOf str) coerce targetType;
+      default = {};
+      example = {
+        acpi_apic_instance = 2;
+        tmem = true;
+        nf_conntrack.acct = 1;
+        acpi_osi = "Windows 2000";
+      };
+      description = let
+        docUrl = "https://www.kernel.org/doc/html/latest/admin-guide/"
+               + "kernel-parameters.html";
+      in ''
+        An attribute set of kernel command line paramaters, where the attribute
+        name denotes a particular parameter. If the attribute value is
+        <literal>true</literal> the attribute name is passed verbatim and if
+        it's <literal>false</literal> the parameter is dropped entirely.
+
+        Have a look at the <link xlink:href="${docUrl}">documentation</link>
+        for possible values.
+      '';
+    };
+
+    boot.initArgs = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = [ "--crash-shell" "--crash-vt=1" ];
+      description = ''
+        Arguments passed to <citerefentry>
+          <refentrytitle>init</refentrytitle>
+          <manvolnum>1</manvolnum>
+        </citerefentry> via kernel parameters.
+
+        You could directly use <option>boot.kernelParams</option> to do the
```

Comment:
This needs `<para>` if you want the paragraph breaks to show up in the docs.

Reply:
I don't mind, but it would probably make sense to do so.